Extrapolate hooks/smart logic into containers to stop mem leaks

W3stside opened the pull request:
# Summary

There was smart logic (hooks, async logic etc) causing memory leaks when these components were being unmounted conditionally inside of SwapMod.

Examples: TradeBasicDetails, RowFee, RowSlippage whom all calculated USD estimation inside their funciton body but were being unmounted inside SwapMod. Now the logic is moved to the relevant parent containers and unmounted inside the render method to safely avoid mem leaks.

  # To Test

1. use this pr and try different prices to show fee warning, then quickly switch amounts
- [ ] in this PR there should be NO errors concerning sth like "can't setState on unmounted component blah blah"
2. Compare this with #1356 where you should see the errors in the console

W3stside committed:
adjust types to accept null

W3stside committed:
conditionally render inside container

W3stside committed:
move some logic into parent containers

W3stside committed:
re-simplify TradePrice

W3stside commented:
not sure why this was merged i didnt merge it. anyways re-opening..

elena-zh commented:
In this PR I see `-100%` or a huge price impact
![-100%](https://user-images.githubusercontent.com/70885163/131500715-6d694171-e6e9-4522-a226-a8fb46b4efb9.jpg)
![100](https://user-images.githubusercontent.com/70885163/131500735-dc2ee901-a3fd-43f4-aa3c-b6aec32956e0.jpg)
![price impact](https://user-images.githubusercontent.com/70885163/131500794-0d621965-84c4-45e2-914f-1f0d383e24fc.jpg)

elena-zh commented:
Btw, sometimes I see these failed requests both in the current and in #1356 
![image](https://user-images.githubusercontent.com/70885163/131501600-2c8d2387-8f6d-4449-8638-92b6f76a4dbb.png)

Is this OK?

W3stside commented:
not related to this PR :) addressing these issues separately

On Tue, Aug 31, 2021 at 1:24 PM Elena ***@***.***> wrote:

> Btw, sometimes I see these failed requests both in the current and in
> #1356 <https://github.com/gnosis/cowswap/pull/1356>
> [image: image]
> <https://user-images.githubusercontent.com/70885163/131501600-2c8d2387-8f6d-4449-8638-92b6f76a4dbb.png>
>
> Is this OK?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/gnosis/cowswap/pull/1364#issuecomment-909188000>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFCY4C6LGPMNMVKYLOVTZNTT7TCXLANCNFSM5DD26B2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
